### DISCUSSION TITLE
Resuming simulation from previous starting point

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#28206] [Q&A Modules: Solid mechanics] [answered]
Hi,

I wanted to know if I can resume a simulation with the deformed mesh of an initial simulation (simple stretching of a plate). From this page (https://mooseframework.inl.gov/modules/porous_flow/restart.html) it mentioned the meshes need to be the same. But after finite biaxial stretching the mesh in the first simulation is a bit different from its original configuration. 

Secondly, I wanted to know if it is possible to restart the aux variables instead of the regular variables. The reason for this is that my first simulation is a simple stretching problem, so my variables are just disp_x and disp_y, but the second one is a frequency domain problem where displacements are split into real and imaginary parts like the last example on this page (https://mooseframework.inl.gov/modules/solid_mechanics/1d_elastic_waves.html).

The other idea I was thinking of is to also split the stretching displacement variables into real and imaginary parts, which would mean applying the displacement BC on the real part only (and setting the imaginary part to 0 on that boundary)? Would that be equivalent?

**Comment by GiudGiud** [comment, marked as answer]
> Secondly, I wanted to know if it is possible to restart the aux variables instead of the regular variables. 

it is possible to do this, but you need to use either:
- exodus restart
- a solutionUO and several solutionIC (for each auxvariable)

I recommend to use Checkpoint restart + the solution UO route

> The other idea I was thinking of is to also split the stretching displacement variables into real and imaginary parts, 

and solving the fully coupled problem? Both displacements and frequency domain at the same time?

**Reply by richmondodufisan** [reply]
For the solutionUserObject route, the syntax is:

```
[UserObjects]
  [./soln]
    type = SolutionUserObject
    mesh = cubesource.e
    system_variables = source_nodal
  [../]
[]
```

Is the `system_variables` the name of the new variable you want to save it in, or the name of the variable from the previous simulation in the exodus file? Or are they supposed to be the same? If so then I might need to rename my displacement variables from the first simulation so I don't mix them up.

Also, the `execute_on` set to `TIMESTEP_END` means that it'll use the last (deformed) shape of the exodus, correct?

Then just to confirm, when I set the solutionIC, it will only use it for the initial condition? I'm asking because in the frequency domain problem, disp_x (and disp_y) are aux variables calculated as sqrt(disp_x_real^2 + disp_x_imag^2). The real and imaginary displacement variables are the ones calculated from the kernel.

***************************************************************************************

The other option I was thinking was to split the disp_x (and disp_y) in the first simulation to disp_x_real and disp_x_imag. Then, on the zero boundary I'd also set both to zero, and on the boundary with a prescribed displacement I'd set disp_x_real to the prescribed value and set disp_x_imag to zero, since that should be equivalent?

Then, for the frequency domain problem that follows it, I would use

`	use_for_exodus_restart = true`

in the mesh block, and then use

```
 [disp_x_real]
    order = SECOND
    family = LAGRANGE
    initial_from_file_var = disp_x_real
    initial_from_file_timestep = LATEST
 []
```

in my variables block (and same for the imaginary and y variables). I was wondering if this approach would be equivalent.

**Reply by GiudGiud** [reply]
system_variables in the UO is just the variables you want to load. So it's from the previous simulation.
The names do not have to match, the solutionIC / solutionAux take care of mapping

> Also, the execute_on set to TIMESTEP_END means that it'll use the last (deformed) shape of the exodus, correct?

 do you really need to execute the user object multiple times? I feel like you just need it for initialization?

Your second option seems fine to me.
I would recommend doing both and weeding out any potential error by comparing the two results

**Reply by richmondodufisan** [reply]
Yes you're right, what I meant to say was to set

`timestep = LATEST`

in the UserObject. Thank you, I'll try them out!
